mne/preprocessing/annotate_amplitude.py

```diff
@@ -0,0 +1,200 @@
+            bads.extend(peak_ch)
+            # add onset/offset for annotations
+            peak_ch_to_annotate = picks_[
+                np.where((0 < peak_mean) & (peak_mean <= bad_percent))[0]]
+            idx = np.where(peak_[peak_ch_to_annotate, :])[1]
+            any_peak[idx] = True
+
+    # annotation for flat
+    annotation_flat = _create_annotations(any_flat, 'flat', raw)
+    # annotation for peak
+    annotation_peak = _create_annotations(any_peak, 'peak', raw)
+    # group
+    annotations = annotation_flat + annotation_peak
+    # bads
+    bads = [bad for bad in bads if bad not in raw.info['bads']]  # sanity-check
+
+    return annotations, bads
+
+
+def _check_ptp(ptp, name, info, picks):
+    """Check the PTP threhsold argument, and converts it to dict if needed."""
+    _validate_type(ptp, ('numeric', dict, None))
+
+    if ptp is not None and not isinstance(ptp, dict):
+        if ptp < 0:
+            raise ValueError(
+                f"Argument '{name}' should define a positive threshold. "
+                f"Provided: '{ptp}'.")
+        ch_types = set(_get_channel_types(info, picks))
+        ptp = {ch_type: ptp for ch_type in ch_types}
+    elif isinstance(ptp, dict):
+        for key, value in ptp.items():
+            if value < 0:
+                raise ValueError(
+                    f"Argument '{name}' should define positive thresholds. "
+                    "Provided for channel type '{key}': '{value}'.")
+    return ptp
+
+
+def _check_bad_percent(bad_percent):
+    """Check that bad_percent is a valid percentage and converts to float."""
+    bad_percent = float(bad_percent)
+    if not 0 <= bad_percent <= 100:
+        raise ValueError(
+            "Argument 'bad_percent' should define a percentage between 0% "
+            f"and 100%. Provided: {bad_percent}%.")
+    return bad_percent
+
+
+def _check_min_duration(min_duration, raw_duration):
+    """Check that min_duration is a valid duration and converts to float."""
+    min_duration = float(min_duration)
+    if min_duration < 0:
+        raise ValueError(
+            "Argument 'min_duration' should define a positive duration in "
+            f"seconds. Provided: '{min_duration}' seconds.")
+    if min_duration >= raw_duration:
+        raise ValueError(
+            "Argument 'min_duration' should define a positive duration in "
+            f"seconds shorter than the raw duration ({raw_duration} seconds). "
+            f"Provided: '{min_duration}' seconds.")
+    return min_duration
+
+
+def _reject_short_segments(arr, min_duration_samples):
+    """
+    Check if flat or peak segments are longer than the minimum duration.
+    """
+    assert arr.dtype == bool and arr.ndim == 2
+    for k, ch in enumerate(arr):
+        onsets, offsets = _mask_to_onsets_offsets(ch)
+        for start, stop in zip(onsets, offsets):
+            if stop - start < min_duration_samples:
+                arr[k, start:stop] = False
+    return arr
```

Comment:
If someone has an idea on how to vectorize this and avoid the for loops.. Feel free to propose 😄 
`arr` shape is `(n_channels, n_samples)` with all channels of the same type.

Reply:
is it critical in running time?

Reply:
Probably not as this is (more or less) the approach used in `flat.py`, which is why I am not spending time on this ;) 
But if someone has a good idea that requires little extra effort.. 😄

Reply:
'little extra effort' 😅 JIT is a good idea, I'll add it once I have the correct code completely tested.

Reply:
The JIT compiled code worked out of the box! Thanks!